certbot/certbot/_internal/main.py

```diff
@@ -1394,6 +1394,18 @@ def run(config: configuration.NamespaceConfig,
     except errors.PluginSelectionError as e:
         return str(e)
 
+    if config.must_staple and installer and "staple-ocsp" not in installer.supported_enhancements():
+        raise errors.NotSupportedError("Must-Staple extension requested, but OCSP stapling "
```

Comment:
```
        raise errors.NotSupportedError(
            "Must-Staple extension requested, but OCSP stapling is not supported by the selected "
            f"installer ({config.installer})\n\n"
            "You can either:\n"
            " * remove the `--must-staple` option from the command line and obtain a certificate "
            "without the Must-Staple extension, or;\n"
            " * use the `certonly` subcommand and manually install the certificate into the  "
            "intended service (e.g. webserver). You must also then manually enable OCSP stapling, "
            "as it is required for certificates with the Must-Staple extension to "
            "function properly.\n"
            " * choose a different installer plugin (such as --nginx or --apache), if possible."
        )
```

Mostly cleaning this up for readability; also added the parenthetical in the third bullet and reworded the second bullet. I'm a little worried that it's too long and people will therefore ignore it, but I think on the whole it's better to have the information for those who want it.